src/pages/AdminPages/ApprovedAccounts/index.js

```diff
@@ -0,0 +1,54 @@
+import React, {useEffect} from 'react'
+import { useSelector, useDispatch, shallowEqual } from 'react-redux'
+import Authenticate from '../../../components/Authenticate'
+import PageNotFound from '../../PageNotFound'
+import { 
+    getTraders, getUsers, getAdmins,
+    getAffiliates, setAlterUsers 
+} from "../../../store/reducers/adminQuery";
+import AlterUsers from '../../../components/Admin/AlterUsers';
+import { setIsLoading } from '../../../store/reducers/user';
+
+const ApprovedAccounts = () => {
+    const dispatch = useDispatch()
+    const { role, isLoading, alterUsers, traders } = useSelector(state => ({
+        role : state.user.role,
+        isLoading : state.user.isLoading,
+        alterUsers: state.adminQuery.alterUsers,
+        traders: state.adminQuery.traders,
+    }), shallowEqual)
+
+    useEffect(() => {
+        dispatch(setIsLoading());
+        dispatch(getUsers())
+        dispatch(getTraders())
+        // eslint-disable-next-line
+    }, [])
+
+    const update = (dispatchFunc) => {
+        dispatch(setIsLoading());
+        dispatch(dispatchFunc())
+    }
+
+    return(
+        <>
+            <Authenticate inside={true} />
+            {
+                role && (
+                    role.isAdmin ? (
+                    <>
+                        <button disabled={alterUsers[0] && alterUsers[0].role.isUser && !alterUsers[0].role.isAffiliate  && true} onClick={() => {update(getUsers)}}>View User Accounts</button> <br />
+                        <button disabled={alterUsers[0] && alterUsers[0].role.isAffiliate  && true} onClick={() => {update(getAffiliates)}}>View Affiliate Accounts</button> <br />
```

Comment:
And what if the first entry in the array is an affiliate which is also in the user list?

I guess my point is (and it's not really important now), disabling the buttons in not neccessary... but we can deal with that later.

Reply:
Well, disabling the button isn't important, but we need to be able to get which button is clicked last so as to know the kind of data that was rendered.

Reply:
So what I did instead os to have a `ROLE` field to let the admin know which category of data is being rendered in the `Table` component.